plugin.go

```diff
@@ -45,6 +46,17 @@ func (p Plugin) Exec() error {
 		}
 
 		destination = path.Base(u.Path)
+	} else if destination[len(destination)-1] == '/' {
```

Comment:
Why this? After getting the destination it should be enough to get the dirname from destination.

Reply:
I think if destination ends with '/' (e.g. `foo/bar/`), we should create dir `foo/bar` and get base name from url, so that file will be downloaded to `foo/bar/<basename>`.

Reply:
I'm not even sure if that really works, that's not a case I had in mind when writing this plugin.

Reply:
I have tested it using CLI.
I think this case is useful too, but the code needs refactoring.